bokeh/plotting/gmap.py

```diff
@@ -230,8 +230,9 @@ def gmap(google_api_key, map_options, **kwargs):
        GMap
 
     '''
-
-    return GMap(api_key=google_api_key, map_options=map_options, **kwargs)
+    encoded_key = Base64String()
+    encoded_key = encoded_key.serialize_value(google_api_key)
```

Comment:
This seems completely unnecessary. It's up to the properties' system to call this method when needed. If I follow the logic correctly, this will result in the API key being encoded twice.

Reply:
I have defined the Base64Encode class to variable 'api_key'. Will passing 'google_api_key' directly to it without explicitly using _'encoded_key.serialize_value(google_api_key)'_ encode the string?

Reply:
@ayushiitr2901 The intention behind all this is that the key can be specified normally on the Python side:
```
return GMap(api_key=google_api_key, map_options=map_options, **kwargs)
```
And that the property system will automatically, transparently encode the value before serializing. This kind of step should not be required of users.

Reply:
Or perhaps a clearer way of putting things: In the making of this PR, *no existing example code should need to be changed in any way*